rdrf/rdrf/services/rest/views/proms_api.py

```diff
@@ -149,6 +149,9 @@ def _update_proms_fields(self, survey_request, survey_data):
         # clinical system
         logger.debug("updating downloaded proms for survey request %s" % survey_request.pk)
         patient_model = survey_request.patient
+        consents = [(c["cde_code"], c["consent_code"]) for c in self.registry_model.metadata.get("consents", [])]
```

Comment:
Instead of an association list ( list of pairs / 2 element list )   )

use a dictionary e.g.

d = { cde_code1:   consent_code1,
        cde_code2:   consent_code2,
   etc }


instead if  iterating

consent_question_code = d.get(cde_code, None)

if consent_question_code is not None:
     # this means is a consent ....